### ISSUE TITLE
v25.1 Release Plan

### ISSUE THREAD

**Issue by eradman** [getredash/redash#7251]
- [x] Create a release branch
- [x] Create new GitHub Release
- [x] Push image to Docker Hub
- [x] Update Redash website
- [x] Announcement in [discussions/6647](https://github.com/getredash/redash/discussions/6647)

References:

[New-version-release-checklist](https://github.com/getredash/redash/wiki/New-version-release-checklist)
[v6 Release Plan](https://github.com/getredash/redash/issues/3015)
[Release Notes](https://github.com/getredash/redash/wiki/Release-Notes)
[Release Candidates and Release Versioning](https://github.com/getredash/redash/discussions/6411)

**Comment by justinclift** [comment]
We'll probably need to make sure the tag that's created doesn't have leading zero's in the month part too.  ie `25.1` vs `25.01`

**Comment by lucydodo** [comment]
Should we write release notes?

**Comment by justinclift** [comment]
@eradman has already been creating release notes in the wiki: https://github.com/getredash/redash/wiki/Release-Notes :smile:

**Comment by eradman** [comment]
> We'll probably need to make sure the tag that's created doesn't have leading zero's in the month part too. ie `25.1` vs `25.01`

Do we want to drop the leading zero? As long as we're consistent in the future either way is fine.

This is is the sequence I have in mind:

```sh
git checkout 25.01.0-dev
git checkout -b v25.1
vim package.json redash/__init__.py pyproject.toml  # set version to 25.1.0
git commit -a -m "25.1.0 release"
git push --set-upstream origin v25.1
gh release create v25.1.0 --target v25.1 --title v25.1.0 --notes "v25.1.0 release"
```

In this scheme any bug fixes and patches can be applied to the `v25.1` branch, followed by a minor release.

**Comment by justinclift** [comment]
> Do we want to drop the leading zero?

Yeah, we definitely do.  It has the potential (and likelihood) of causing issues with things that can't deal with leading zeros, and we don't really need to take that chance.

@arikfr Pointed out a while ago that it'll probably cause issues with older versions of Redash that are doing version checks too, so we should fix the problem before the next release.

> This is is the sequence I have in mind ...

Looks good to me, nothing is immediately jumping out as wrong with that. :smile:

**Comment by eradman** [comment]
Created the v25.1 branch: https://github.com/getredash/redash/tree/v25.1

Next steps I think are to create a Github release and deploy and image to Dockerhub

**Comment by eradman** [comment]
Created a release, we can edit the description as needed

![v25 1 0-release](https://github.com/user-attachments/assets/e86d861d-6cb4-440a-80d8-b0583a4ae83b)

Next step is to publish a Docker image

**Comment by eradman** [comment]
Manually ran the Preview Image action, which succeeded in pushing the 25.1.0 image:

https://hub.docker.com/layers/redash/redash/25.1.0/images/sha256-f1eb5a2529cfcddb35d7cec37858441557620164e9899d3dd0692073769f7fd3

Pulling this image to see how it works!

**Comment by lucydodo** [comment]
I just merged https://github.com/getredash/website/pull/812. It seems like we've completed all the checklist items—can we go ahead and close it?
Also, I want to take a moment to thank everyone who contributed to this release. :)

**Comment by eradman** [comment]
>  It seems like we've completed all the checklist items—can we go ahead and close it?

I think so. If we'll start a new ticket for a bugfix-only minor release.

> Also, I want to take a moment to thank everyone who contributed to this release. :)

I concur!

**Comment by arikfr** [comment]
Congrats and thank you everyone who took part in this 🙏 🎉

**Comment by ericminio** [comment]
Hi
Is the https://github.com/getredash/setup still relevant for this version?

**Comment by ericminio** [comment]
Nevermind, works well!
thx

**Comment by snickerjp** [comment]
@eradman @arikfr @justinclift 

# Updating the `latest` Tag of the Docker Image  

Congratulations on the release of Version 25.1.0! 🎉  

The `latest` tag is described as pointing to the most recent stable version, but it doesn't seem to be updated to the current version.  
Do you have any plans to update this tag to the latest version?  

For reference, here is the current `latest` tag:  
[Docker Hub - `redash/redash:latest`](https://hub.docker.com/layers/redash/redash/latest/images/sha256-55cedc91a09107e894a0ce872d138ec8c708ec537f6df282207bdf14eec72491)  

Also, if there is an official policy regarding Docker tag updates, I would appreciate it if you could share the reference.  

![Image](https://github.com/user-attachments/assets/37b74c7d-a932-475c-9114-a2121ade49f7)
